src/Docker/Runner/DataLoader/ExternallyManagedWorkspaceCredentials.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DockerBundle\Docker\Runner\DataLoader;
+
+use Keboola\DockerBundle\Exception\ExternalWorkspaceException;
+use Keboola\StagingProvider\Provider\Credentials\DatabaseWorkspaceCredentials;
+
+readonly class ExternallyManagedWorkspaceCredentials
+{
+    private function __construct(
+        public string $id,
+        public string $type,
+        public string $password,
+    ) {
+    }
+
+    public static function fromArray(array $credentials): self
+    {
+        if (!isset($credentials['id']) || !isset($credentials['type']) || !isset($credentials['#password'])) {
+            throw new ExternalWorkspaceException(
+                'Missing required fields (id, type, #password) in workspace_credentials',
+            );
+        }
+        if ($credentials['type'] !== 'snowflake') {
+            throw new ExternalWorkspaceException(sprintf('Unsupported workspace type "%s"', $credentials['type']));
+        }
```

Comment:
Tady to uz je za validaci configu, ne? Takze ty to nemelo byt `UserException`, ale normalni `InvalidArgument`

Reply:
Pravdu díš

Reply:
To by byla škoda mazat tak krásnou výjimku